Tests/SystemTests/MachPortTests.swift

```diff
@@ -110,34 +155,84 @@ final class MachPortTests: XCTestCase {
     func testMakeSendOnceDoesntIncrementMakeSendCount() throws {
         let recv = Mach.Port<Mach.ReceiveRight>()
         let zero = recv.makeSendCount
-        XCTAssert(zero == 0)
+        XCTAssertEqual(zero, 0)
         _ = recv.makeSendOnceRight()
         let same = recv.makeSendCount
-        XCTAssert(same == zero)
+        XCTAssertEqual(same, zero)
     }
 
     func testMakeSendOnceIsUnique() throws {
         let recv = Mach.Port<Mach.ReceiveRight>()
         let once = recv.makeSendOnceRight()
         recv.withBorrowedName { rname in
             once.withBorrowedName { oname in
-                print(oname, rname)
-                XCTAssert(oname != rname)
+                XCTAssertNotEqual(oname, rname)
             }
         }
     }
 
     func testCopySend() throws {
         let recv = Mach.Port<Mach.ReceiveRight>()
         let zero = recv.makeSendCount
-        XCTAssert(zero == 0)
+        XCTAssertEqual(zero, 0)
         let send = recv.makeSendRight()
         let one = recv.makeSendCount
-        XCTAssert(one == 1)
+        XCTAssertEqual(one, 1)
         _ = try send.copySendRight()
         let same = recv.makeSendCount
-        XCTAssert(same == one)
+        XCTAssertEqual(same, one)
+
+    }
+
+    func testCopyDeadName() throws {
+        let recv = Mach.Port<Mach.ReceiveRight>()
+        let send = recv.makeSendRight()
+        _ = consume recv // and turn `send` into a dead name
+        do {
+            let copy = try send.copySendRight()
+            _ = copy
+        }
+        catch Mach.PortRightError.deadName {
+          // success
+        }
+    }
+
+    func testCopyDeadName2() throws {
+        let send = Mach.Port<Mach.SendRight>(name: 0xffffffff)
+        do {
+            let copy = try send.copySendRight()
+            _ = copy
+        }
+        catch Mach.PortRightError.deadName {
+          // success
+        }
```

Comment:
Oh actually this doesn't work because `XCTAssertThrowsError` is generic over the return type of the expression, but noncopyable types cannot be used in a generic context at this time.

Reply:
Got it to work by making the expression return `Void` instead.